WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java

```diff
@@ -404,169 +533,67 @@ private JSONObject serializeDotComParamsToJSONObject() throws JSONException {
-                    }
-                });
-    }
-
-    private void updateJetpackMonitorSettings() {
-        WordPress.getRestClientUtils().setJetpackMonitor(
-                mSite.getSiteId(), mJpSettings.monitorActive, new RestRequest.Listener() {
-                    @Override
-                    public void onResponse(JSONObject response) {
-                        mRemoteJpSettings.monitorActive = response.optBoolean("active");
-                        mJpSettings.monitorActive = mRemoteJpSettings.monitorActive;
-                        notifySavedOnUiThread(null);
-                    }
-                }, new RestRequest.ErrorListener() {
-                    @Override
-                    public void onErrorResponse(VolleyError error) {
-                        AppLog.w(AppLog.T.API, "Error updating Jetpack Monitor settings: " + error);
-                    }
-                });
-    }
-
-    private void saveJetpackSettings() {
-        updateJetpackMonitorSettings();
-        final Map<String, String> params = serializeJetpackParams();
-        if (params == null || params.isEmpty()) return;
-
-        WordPress.getRestClientUtils().setJetpackSettings(
-                mSite.getSiteId(), new RestRequest.Listener() {
-                    @Override
-                    public void onResponse(JSONObject response) {
-                        AppLog.d(AppLog.T.API, "Jetpack Settings saved remotely");
-                        mRemoteJpSettings.emailNotifications = mJpSettings.emailNotifications;
-                        mRemoteJpSettings.wpNotifications = mJpSettings.wpNotifications;
-                        notifySavedOnUiThread(null);
-                    }
-                }, new RestRequest.ErrorListener() {
-                    @Override
-                    public void onErrorResponse(VolleyError error) {
-                        AppLog.w(AppLog.T.API, "Error POSTing Jetpack site settings changes: " + error);
-                        notifySavedOnUiThread(error);
-                    }
-                }, params);
-    }
-
     private void deserializeJetpackRestResponse(SiteModel site, JSONObject response) {
         if (site == null || response == null) return;
         JSONObject settingsObject = response.optJSONObject("settings");
         mRemoteJpSettings.emailNotifications = settingsObject.optBoolean(JP_MONITOR_EMAIL_NOTES_KEY, false);
         mRemoteJpSettings.wpNotifications = settingsObject.optBoolean(JP_MONITOR_WP_NOTES_KEY, false);
     }
 
-    private Map<String, String> serializeJetpackParams() {
+    private @NonNull Map<String, String> serializeJetpackMonitorParams() {
         Map<String, String> params = new HashMap<>();
-        params.put(JP_MONITOR_EMAIL_NOTES_KEY, String.valueOf(mJpSettings.emailNotifications));
-        params.put(JP_MONITOR_WP_NOTES_KEY, String.valueOf(mJpSettings.wpNotifications));
+        if (mJpSettings.emailNotifications != mRemoteJpSettings.emailNotifications) {
```

Comment:
We're using these checks to prevent unnecessary network calls (see `pushJetpackProtectAndSsoSettings` and `pushWpSettings`). I think we can go about preventing those calls just using `equals` which would allow us to remove these checks, I'd prefer to tackle that in a separate PR.